File: tasks/model/CarpoolGroupPost.py
from mongoengine import *
from datetime import datetime

class CarpoolGroupPost(Document):

    post_id = LongField(required=True)
    post_url = URLField(required=True)
    title = StringField()
    price = StringField()
    content = StringField()
    photos = ListField(URLField())
    location = StringField()
    created_at = LongField()
    collected_at = LongField(default=lambda: datetime.utcnow().timestamp())

File: server/lib/models/CarpoolGroupPost.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const CarpoolGroupPostSchema = new Schema({
  post_id: {
    type: Number,
    required: true,
  },
  post_url: {
    type: String,
    required: true,
  },
  title: {
    type: String,
  },
  price: {
    type: String,
  },
  content: {
    type: String,
  },
  photos: {
    type: [String]
  },
  location: {
    type: String,
    required: true,
  },
  created_at: {
    type: Number,
    required: true,
  }
}, { collection: 'carpool_group_post' });

module.exports = mongoose.model('CarpoolGroupPost', CarpoolGroupPostSchema);

